anx/provider/instances.go

```diff
@@ -56,30 +58,39 @@ func (i instanceManager) NodeAddressesByProviderID(ctx context.Context, provider
 	return nodeAddresses, nil
 }
 
-func (i instanceManager) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
+func (i *instanceManager) handleUnauthorizedForbidden(err error) {
+	if utils.IsUnauthorizedOrForbiddenError(err) {
+		i.lastUnauthorizedOrForbiddenInstanceExistCall = time.Now()
+	}
+}
+
+func (i *instanceManager) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
+	if i.lastUnauthorizedOrForbiddenInstanceExistCall.Add(time.Minute).After(time.Now()) {
+		return false, utils.ErrUnauthorizedForbiddenBlocked
+	}
```

Comment:
is this really the only method we need this check in? What about e.g. `InstanceMetadata()`?

Reply:
`InstanceMetadata` has two call paths. One has an exponential backoff and the other is every five minutes. `InstanceShutdown` (the only other Instance method consumed by the runtime) is only called if `InstanceExist` returns with no errors.